static/js/util/profile_edit.js

```diff
@@ -162,6 +164,9 @@ export function boundSelectField(keySet, label, options) {
     _.set(clone, editKeySet, undefined);
 
     updateProfile(clone);
+    if (onChange !== undefined) {
```

Comment:
I prefer runtime errors for those cases since there's a clear logic error on our part if we're passing a non-function as the onChange argument and a runtime error might make that easier to diagnose. But in this case it probably doesn't matter, an error on our part would mean the country field would not be synced up with the state field anymore, which would be preferable to a runtime error halting the render method